### ISSUE TITLE
Swiss feeds

### ISSUE THREAD

**Issue by drewda** [transitland/transitland#52]
http://gtfs.geops.ch/

- [ ] @doublestranded imports on dev stack and looks for RSP errors
- [ ] try to find a local who can help us review and edit operator names and metro/region names
- [ ] add and import to production

**Comment by doublestranded** [comment]
The Feed managed to import stops and routes after increasing the dev worker instance memory up to 64 GiB (it's now back at 8 GiB); however SSPs didn't import most likely because the database server can't handle the load. Also might be issues with rails controller/serializer.

**Comment by doublestranded** [comment]
The import and worker memory used never seemed to go above 16 GiB. As a feed, there are no route shapes (they are auto-generated from the stops) and the stop_times file is almost 400 MB

**Comment by batje** [comment]
@stevevance  said: "The Switzerland feed in changeset 673 has 409 operators. That's insane, and that's going to take a very long time for me, or anyone, to edit. " in #116 

I checked a bit. And insane as it may be, it appears to reflect real life. Here is a few agents, googled from the file:

In English:
https://en.wikipedia.org/wiki/Wynental_and_Suhrental_railway

In dutch:
https://nl.wikipedia.org/wiki/TRAVYS
TRAVYS is actually a merger, so you are lucky, they have 3 train routes and 3 bus routes.

But (and this took me a bit) the main reason might be that all the ski lift operators are privately owned public transport companies. 
Like this one:
https://de.wikipedia.org/wiki/Davos_Klosters_Bergbahnen
DKB-bbbj (Davos Klosters Bergbahnen (bbbj))
They own 3 smaller companies, at least one of them is still in the file with its original name: SBP (Sportbahnen Pischa)

The Swiss alps are beautiful, we should all visit one day.

**Comment by drewda** [comment]
Responding to @stevevance's question from https://github.com/transitland/transitland/issues/116#issuecomment-244785285 over on this ticket:

No, rerunning the Feed Registry submission process would not populate `country`/`state`/`metro` values for the operators. At present, those values have to be manually set (although in the future we could potentially automate it using the Who's on First gazetteer https://github.com/whosonfirst/go-whosonfirst-breaches/issues/2 )

I suspect this will come down to us finding someone in Switzerland who's familiar with the language(s) and regions and is willing to help us with the cataloging. As @batje is also noting, it's complex.

**Comment by irees** [comment]
ian todo: stops per agency_id

**Comment by stevevance** [comment]
I created a changeset for seven Swiss operators and applied it. [You can see the 7 on the Feed Registry.](https://transit.land/feed-registry/?country=Switzerland)

**Comment by irees** [comment]
Level 1 import completed for Switzerland. However, a more recent FeedVersion has an issue with a missing Stop that is referenced from a transfer but not imported. https://github.com/transitland/transitland-datastore/issues/985

**Comment by drewda** [comment]
from @stevevance:

Someone [replied](https://twitter.com/chtrachsel/status/839235366825701384) to the Transitland Twitter account with a link to some another source for a Swiss feed. 

https://opentransportdata.swiss/en/dataset/timetable-2017-gtfs

I don't know if this is different from our current source on http://gtfs.geops.ch/

**Comment by slvlirnoff** [comment]
http://gtfs.geops.ch/ is an un-official release based on the transformation of an official non-gtfs datasource.
opentransportdata.swiss/... is the official GTFS feed, and from my understanding it seems of higher quality (not using a route per trip, etc.) and updated more often.

In order to import & maintain it more easily https://github.com/transitland/transitland-datastore/issues/1073 would help because https://opentransportdata.swiss/en/dataset/timetable-2017-gtfs/permalink is a zip with an internal structure that will change on every update (it contains the date of the version).

**Comment by stevevance** [comment]
@slvlirnoff The way things work on Transitland right now is that every time the folder structure changes, we have to update the subdirectory.

**Comment by slvlirnoff** [comment]
@stevevance yes I know and that's why I linked to https://github.com/transitland/transitland-datastore/issues/1073 which would make it work if there's only one non ambiguous way to read the zip (i.e. only one feed in the archive).

This improvement would make the official swiss feed permalink a stable url and make it possible to get the updates without a fetch source invalid error.

**Comment by stevevance** [comment]
The status of this is that we've only imported the initial 7 operators back in January. There are 390 operators that haven't been imported. 

[The feed](https://transit.land/dispatcher/feeds/f-u0-switzerland) is set to `manual_import: true`. 

@irees Can I set this to import level 4?
